LFPy/recextelectrode.py

```diff
@@ -406,153 +470,54 @@ def _loop_over_contacts(self, **kwargs):
-                np.random.seed(self.seedvalue)
-
-            if self.contact_shape is 'circle':
-                for j in range(self.n):
-                    A = [(np.random.rand()-0.5)*self.r*2,
-                        (np.random.rand()-0.5)*self.r*2,
-                        (np.random.rand()-0.5)*self.r*2]
-                    offs[j, ] = np.cross(self.N[i, ], A)
-                    r2[j] = offs[j, 0]**2 + offs[j, 1]**2 + offs[j, 2]**2
-                    while r2[j] > self.r**2:
-                        A = [(np.random.rand()-0.5)*self.r*2,
-                            (np.random.rand()-0.5)*self.r*2,
-                            (np.random.rand()-0.5)*self.r*2]
-                        offs[j, ] = np.cross(self.N[i, ], A)
-                        r2[j] = offs[j, 0]**2 + offs[j, 1]**2 + offs[j, 2]**2
-            elif self.contact_shape is 'square':
-                for j in range(self.n):
-                    A = [(np.random.rand()-0.5),
-                        (np.random.rand()-0.5),
-                        (np.random.rand()-0.5)]
-                    offs[j, ] = np.cross(self.N[i, ], A)*self.r
-                    r2[j] = offs[j, 0]**2 + offs[j, 1]**2 + offs[j, 2]**2
-
-            x_n = offs[:, 0] + self.x[i]
-            y_n = offs[:, 1] + self.y[i]
-            z_n = offs[:, 2] + self.z[i]
-
-            return x_n, y_n, z_n
-
-        def loop_over_points(x_n, y_n, z_n):
-
-            #loop over points on contact
+        def loop_over_points(points):
+
+            # loop over points on contact
+            lfp_e = 0
             for j in range(self.n):
                 tmp = self.lfp_method(self.cell,
-                                              x = x_n[j],
-                                              y = y_n[j],
-                                              z = z_n[j],
-                                              r_limit = self.r_limit,
-                                              sigma = self.sigma,
-                                              **kwargs
-                                              )
-
-                if j == 0:
-                    lfp_e = tmp
-                else:
-                    lfp_e += tmp
-
-                #no longer needed
+                                      x=points[j, 0],
+                                      y=points[j, 1],
+                                      z=points[j, 2],
+                                      r_limit=self.r_limit,
+                                      sigma=self.sigma,
+                                      **kwargs
+                                      )
+
+                lfp_e += tmp
+                # no longer needed
                 del tmp
 
             return lfp_e / self.n
 
-        #loop over contacts
-        for i in range(len(self.x)):
-            if self.n > 1:
-
-                #fetch offsets:
-                x_n, y_n, z_n = calc_xyz_n(i)
-
+        # extract random points for each electrode
```

Comment:
maybe I can ?save the recorded points in `self.recorded_points` or something like that?

Reply:
Is that really needed for anything? We don't return the tmp potentials at each point, only the averaged potential

Reply:
I don't know..it could be potentially interesting to compare if the position of the random points matter or something like that. I don't think it's harmful to have it :)

Reply:
I suggest we add these positions in a later PR when and if someone feel the need to investigate it. I think I did some analysis of predicted LFPs for different seeds - the predictions are very similar if the n-parameter is sufficiently large. And you don't need to look at the exact locations to compute these distributions (the distribution of amplitudes for different instantiations at a given time step given a choice of n).